codex-rs/exec/src/event_processor_with_human_output.rs

```diff
@@ -168,9 +179,15 @@ impl EventProcessor for EventProcessorWithHumanOutput {
             EventMsg::BackgroundEvent(BackgroundEventEvent { message }) => {
                 ts_println!(self, "{}", message.style(self.dimmed));
             }
-            EventMsg::TaskStarted | EventMsg::TaskComplete(_) => {
+            EventMsg::TaskStarted => {
                 // Ignore.
             }
+            EventMsg::TaskComplete(TaskCompleteEvent { last_agent_message }) => {
+                return handle_last_message(
+                    last_agent_message.as_deref(),
+                    self.last_message_path.as_deref(),
+                );
```

Comment:
And then I would do this:

```suggestion
                handle_last_message(
                    last_agent_message.as_deref(),
                    self.last_message_path.as_deref(),
                );
                return CodexStatus::InitiateShutdown;
```